Cargo.lock

```diff
@@ -15,9 +15,9 @@ dependencies = [
 
 [[package]]
 name = "addr2line"
-version = "0.21.0"
+version = "0.22.0"
 source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "8a30b2e23b9e17a9f90641c7ab1549cd9b44f296d3ccbf309d2863cfe398a0cb"
+checksum = "6e4503c46a5c0c7844e948c9a4d6acd9f50cccb4de1c48eb9e291ea17470c678"
```

Comment:
Did you run `cargo update` or regenerated `Cargo.lock` from scratch?
Looks like much more dependencies were updated than strictly necessary.

While with `crates.io` dependencies it should be generally safe, I would still prefer to not mix `cargo update` with other potentially breaking changes.

Would it be possible to restore the original `Cargo.lock` both here and for prover, and then only apply the required changes (I think `cargo` will do it itself on build)?